httpx/middleware.py

```diff
@@ -158,3 +149,17 @@ def redirect_content(self, request: AsyncRequest, method: str) -> bytes:
         if request.is_streaming:
             raise RedirectBodyUnavailable()
         return request.content
+
+
+def basic_auth_str(
+    username: typing.Union[str, bytes], password: typing.Union[str, bytes]
+) -> str:
+    if isinstance(username, str):
+        username = username.encode("latin1")
+
+    if isinstance(password, str):
+        password = password.encode("latin1")
```

Comment:
Out of curiosity, why defaults to ‘latin1’?

Reply:
Technically the RFC says we can use any encoding as long as it's US-ASCII compatible (recommends `latin1` and `utf-8`) so we can switch to UTF-8 here and still be compliant. I'm almost leaning towards we should switch, UTF-8 is ubiquitous. What do others think?

Reply:
Yes, utf-8 probably makes sense given the spec there.

An alternative: We *could* potentially defer to whatever `request.headers.encoding` is set as (we wouldn't be able to build this string at the point of `__init__`, but that's okay.) We would probably then want our logic to be something like...

```python
if request.headers.encoding in ('ascii', 'utf-8'):
    encoding = 'utf-8'
else:
    encoding = request.headers.encoding
```

I'm not convinced that'd be better overall tho - just sticking with utf-8 in this case seems reasonable.

Reply:
Let's use utf-8, and reference the `RFC 7617 Section 2 link in a code comment, yeah?
If users need something else then they can use a customized auth class/middleware instead.